### PR TITLE
Release action for Carpet Extra and bumping Carpet core version

### LINKED ISSUE #166: Update for minecraft 1.6.5
Is there an updated version for 1.16.5

### LINKED ISSUE #172: 1.16.5 Version??
Are their any plans to upgrade this to 1.16.5 so that it will work with Carpet?

### PR DESCRIPTION
**Pull request by altrisi**
_**New:** As for the Carpet dependency bump, it's just because the new Rules-to-wiki uses a field and noop from it and else the build would fail, the PR look bad, and bad things may happen. Users should still be able to use older core versions with this though, since nothing else references that class. Nothing else on that regard, just the bump. Now as for the Release action:_

This PR is basically the equivalent to gnembon/fabric-carpet#650 for Carpet Extra. Only differences with that one will be highlighted in this PR, except for general things.

Fixes _people complaining about Curseforge being outdated_.
This actually happens quite a lot for extra (last update in CF was in August, poor people), for example, this fixes #172 and fixes #166, but IIRC there were some others related to this, [plus tens of comments in CF](https://www.curseforge.com/minecraft/mc-mods/carpet-extra#comments).

Fixes _Rules to Readme making a mess of commits that have to be pulled all the time_.

Fixes _people not finding rules because they haven't been released yet_.

Now, apart from automatically building when a release is pushed as configured in the `gradle.properties` file, releasing it to Github and releasing it to Curseforge, what _extras_ does this have?

Since there are no Scarpet docs, that job isn't ported, and since here the rules are sent to the readme instead of the wiki, the current rules to readme action has been adapted to fit into the publish release instead, to enjoy only having the bot commit when releasing, not on every push to `CarpetExtraSettings`.

This PR at the same time moves the `RulePrinter.java` into the source, as it was done in Carpet some time ago.

And, since Carpet Extra is not published to maven (AFAIK), the publishing step and uploading to the action is skipped.

Now, as in the other PR, this needs the `CF_API_TOKEN` secret to be set before it will work. The CF release has an extra `relations` field to mark Carpet as a required dependency. CF release is ofc not tested, but should work in theory, since it's basically the same as Carpet's barring id and name.

Previews: We have two:
- The almost latest iteration of the action, completely working except for Rules-to-readme (silently fails) ([Release](https://github.com/altrisi/carpet-extra/releases/tag/test2) | [Action run](https://github.com/altrisi/carpet-extra/actions/runs/535083714))
- The (no longer) latest ~(and included)~ iteration of the action, with Rules-to-Readme failing only because it's expecting `RulePrinter` to be on the `master` branch, which isn't there yet ([Release](https://github.com/altrisi/carpet-extra/releases/tag/test3) | [Action run](https://github.com/altrisi/carpet-extra/actions/runs/535142061))
- The actually latest (and included except for one line required for testing) iteration of the action, with the latest and future-proof Rules-to-wiki ([Release](https://github.com/altrisi/carpet-extra/releases/tag/test5) | [Action run](https://github.com/altrisi/carpet-extra/actions/runs/558554423))

**NOTE:** _This PR does NOT include automatic versioning for the `version` field in the `fabric.mod.json`. That still has to be done manually, unless it's changed in another commit._

### CODE REVIEW

**Comment by altrisi** [comment]
Ok, hold this for a moment, actions/upload-release-asset#78 (it is being discontinued rn). I'll have to update it to a different action.

**Comment by altrisi** [comment]
Well, this is now ready, but rules-to-readme won't work/build until gnembon/fabric-carpet#696 is merged, published to maven, and Carpet Extra's `gradle.properties` updated to use that version.

**Comment by altrisi** [comment]
Actually, this is still wrong. Extensions must start on ModInitializer or earlier, DedicatedServerModInitializer is where Carpet already starts.

**Review by gnembon — APPROVED** [review]

**Pull request merged**
